ts/Data/Modifiers/RangeModifier.ts

```diff
@@ -128,99 +125,19 @@ class RangeModifier extends DataModifier {
-                i < iEnd;
-                ++i
-            ) {
-                range = ranges[i];
-
-                if (
-                    strict &&
-                    typeof range.minValue !== typeof range.maxValue
-                ) {
-                    continue;
-                }
-
-                if (i > 0 && !additive) {
-                    modified.deleteRows();
-                    modified.setRows(rows);
-                    modified.setOriginalRowIndexes(indexes, true);
-                    columns = modified.getColumns();
-                    rows = [];
-                    indexes = [];
-                }
-
-                rangeColumn = (columns[range.column] || []);
-
-                for (
-                    let j = 0,
-                        jEnd = rangeColumn.length,
-                        cell: DataTable.CellType,
-                        row: DataTable.Row | undefined,
-                        originalRowIndex: number | undefined;
-                    j < jEnd;
-                    ++j
-                ) {
-                    cell = rangeColumn[j];
-
-                    switch (typeof cell) {
-                        default:
-                            continue;
-                        case 'boolean':
-                        case 'number':
-                        case 'string':
-                            break;
-                    }
-
-                    if (
-                        strict &&
-                        typeof cell !== typeof range.minValue
-                    ) {
-                        continue;
-                    }
-
-                    if (
-                        cell >= range.minValue &&
-                        cell <= range.maxValue
-                    ) {
-                        if (additive) {
-                            row = table.getRow(j);
-                            originalRowIndex = table.getOriginalRowIndex(j);
-                        } else {
-                            row = modified.getRow(j);
-                            originalRowIndex = modified.getOriginalRowIndex(j);
-                        }
-
-                        if (row) {
-                            rows.push(row);
-                            indexes.push(originalRowIndex);
-                        }
-                    }
-                }
-            }
-
-            modified.deleteRows();
-            modified.setRows(rows);
-            modified.setOriginalRowIndexes(indexes);
-        }
+
+        let { start, end } = modifier.options;
+        start = Math.max(0, start || 0);
+        end = Math.min(end || Infinity, table.getRowCount());
+        const length = Math.max(end - start, 0);
+
+        const modified = table.modified;
+        modified.deleteRows();
+        modified.setRows(table.getRows(start, length));
+        modified.setOriginalRowIndexes(Array.from(
+            { length },
+            (_, i): number => i + start
+        ));
```

Comment:
A self-comment so I don't forget later - I should probably use the IDs from the parent here instead of creating a new array. To be fixed.